pkg/apis/config/v1alpha1/interpretercontext_types.go

```diff
@@ -153,6 +153,12 @@ type DependentObjectReference struct {
 	Namespace string `json:"namespace,omitempty"`
 
 	// Name represents the name of the referent.
-	// +required
-	Name string `json:"name"`
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// A label query over a set of resources.
+	// If name is not empty, labelSelector will be ignored.
+	// Name and LabelSelector cannot be empty at the same time.
+	// +optional
+	LabelSelector string `json:"labelSelector,omitempty"`
```

Comment:
Should be `LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"``?

Reply:
+1

And maybe we need to add the check logic when get the response from the interpreter webhook.

Reply:
> Should be LabelSelector *metav1.LabelSelector json:"labelSelector,omitempty"``?


I have also considered using this approach, but I found that it may not be very friendly to Lua. When developing with Lua, you need to understand the structure of *metav1.LabelSelector.

Reply:
In fact, we have already been using more complex structures, such as:
https://github.com/karmada-io/karmada/blob/d27faeac953043c2274bb8787bbbaaafa7d0dcd6/pkg/resourceinterpreter/customized/declarative/luavm/lua.go#L127

We can consider providing some Lua templates in the future to facilitate users' reuse.

Reply:
>  requires = &workv1alpha2.ReplicaRequirements{} 


Ooooooh, thanks reminder.